### PR TITLE
Dialyzer fixes.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Address a series of dialyzer bugs.  Some, but not all.  @seancribbs

### CODE REVIEW

File: src/riak_core_cluster_mgr.erl [hunk 1]
```diff
@@ -108,7 +108,6 @@
 %%% API
 %%%===================================================================
 %% @doc start the Cluster Manager
--spec(start_link() -> ok).
```
Comment (seancribbs) [inline src/riak_core_cluster_mgr.erl:111]: I realize these specs are wrong, but is there a good reason to remove them from API functions?

File: src/riak_core_service_mgr.erl [hunk 2]
```diff
@@ -117,14 +117,12 @@ start_link({IP,Port}) when is_integer(Port), Port >= 0 ->
 %% for taking ownership of the socket via
 %% Transport:controlling_process(Socket, Pid).  Only the strategy of
 %% `round_robin' is supported; it's arg is ignored.
--spec(register_service(hostspec(), service_scheduler_strategy()) -> ok).
```
Comment (seancribbs) [inline src/riak_core_service_mgr.erl:120]: Ditto comment re: removal of specs from API functions

File: src/riak_repl2_fscoordinator.erl [hunk 3]
```diff
@@ -314,8 +314,8 @@ handle_cast(start_fullsync,  State) ->
 
 handle_cast(stop_fullsync, State) ->
     % exit all running, cancel all timers, and reset the state.
-    [erlang:cancel_timer(Tref) || {_, {_, Tref}} <- State#state.whereis_waiting],
-    [begin
+    _ = [erlang:cancel_timer(Tref) || {_, {_, Tref}} <- State#state.whereis_waiting],
+    _ = [begin
```
Comment (seancribbs) [inline src/riak_repl2_fscoordinator.erl:318]: I'd make a minor note to extract this into another function for clarity.

File: src/riak_repl2_ip.erl [hunk 4]
```diff
@@ -11,7 +11,6 @@
 %% @doc Given the result of inet:getifaddrs() and an IP a client has
 %%      connected to, attempt to determine the appropriate subnet mask.  If
 %%      the IP the client connected to cannot be found, undefined is returned.
--spec determine_netmask(Ifaddrs :: [{atom(), any()}], SeekIP :: string() | {integer(), integer(), integer(), integer()}) -> 'undefined' | binary().
```
Comment (seancribbs) [inline src/riak_repl2_ip.erl:14]: Again, API function. Might be better to simply use the proper types, e.g. `inet:ip_address()`

**Comment by seancribbs** [comment]
Now that the shoe is on the other foot... I think it would be more helpful to put the warnings that were removed in the extended commit message. Then I can see that your change actually addressed the original warning. On the whole though, this is a vast improvement, I count 89 warnings removed.

File: src/riak_core_cluster_mgr.erl (hunk 1 above)
Comment (cmeiklejohn) [inline src/riak_core_cluster_mgr.erl:111]: I put in the commit description that the goals was to re-add the signatures with typer once dialyzer was green.

Comment (seancribbs) [inline src/riak_core_cluster_mgr.erl:111]: Cool, thanks.

**Comment by seancribbs** [comment]
:+1: so far. Are you going to continue eliminating warnings or do we want this to merge?

**Comment by cmeiklejohn** [comment]
Going to merge and start a new PR moving forward, does that work?

**Comment by seancribbs** [comment]
Works for me.

**Pull request merged**